xarray/core/computation.py

```diff
@@ -920,6 +920,11 @@ def earth_mover_distance(first_samples,
 
     if input_core_dims is None:
         input_core_dims = ((),) * (len(args))
+    if len(input_core_dims) != len(args):
```

Comment:
This could use `elif` instead -- there's no need to check this if `input_core_dims` was just assigned.